gadget/install/install.go

```diff
@@ -227,37 +226,73 @@ func installOnePartition(dgpair *gadget.OnDiskAndGadgetStructurePair,
 
 // resolveBootDevice auto-detects the boot device
 // bootDevice forces the device. Device forcing is used for (spread) testing only.
-func resolveBootDevice(bootDevice string, bootVol *gadget.Volume) (string, error) {
+func resolveBootDevice(bootDevice string, bootVol gadget.DeviceVolume) (string, error) {
 	if bootDevice != "" {
 		return bootDevice, nil
 	}
+
+	// XXX: It makes no sense currently do handle bootVol.Device here and
+	// force the assignment there, as current constraints dictate the boot
+	// device must be on the same disk as ubuntu-seed. Therefor we should
+	// just ensure that the two disk paths don't differ if assigned
+
+	// default behavior for unassigned volumes
 	foundDisk, err := disks.DiskFromMountPoint("/run/mnt/ubuntu-seed", nil)
 	if err != nil {
 		logger.Noticef("Warning: cannot find disk from mounted seed: %s", err)
 	} else {
 		return foundDisk.KernelDeviceNode(), nil
 	}
-	bootDevice, err = diskWithSystemSeed(bootVol)
+	bootDevice, err = diskWithSystemSeed(bootVol.Volume)
 	if err != nil {
 		return "", fmt.Errorf("cannot find device to create partitions on: %v", err)
 	}
-
+	if bootVol.Device != "" {
+		// disk is assigned for this volume, then it must match the one
+		// that is assigned to system-seed
+		bootDisk, err := disks.DiskFromDeviceName(bootVol.Device)
+		if err != nil {
+			return "", err
+		}
+		logger.Noticef("volume %s has been assigned disk %s", bootVol.Volume.Name, bootDisk.KernelDeviceNode())
+		if bootDevice != bootDisk.KernelDeviceNode() {
+			return "", fmt.Errorf("volume %s was assigned disk %s, but this does not match the disk for ubuntu-seed %s",
+				bootVol.Volume.Name, bootDisk.KernelDeviceNode(), bootDevice)
+		}
+	}
 	return bootDevice, nil
 }
 
+func stripDeviceVolumes(volumes map[string]gadget.DeviceVolume) map[string]*gadget.Volume {
+	stripped := make(map[string]*gadget.Volume)
+	for name, vol := range volumes {
+		stripped[name] = vol.Volume
+	}
+	return stripped
+}
+
+func findBootVolume(volumes map[string]gadget.DeviceVolume) (string, error) {
```

Comment:
maybe `bootVolumeName`

Reply:
Nice suggestion, thank you, naming suggestions are always welcome